chain/xrp/client/client.go

```diff
@@ -0,0 +1,906 @@
+		Fee:           xc.NewAmountBlockchainFromStr(txResponse.Result.Fee),
+		BlockIndex:    txResponse.Result.LedgerIndex,
+		BlockTime:     txResponse.Result.Date,
+		Confirmations: confirmations,
+		Status:        status,
+		Sources:       sources,
+		Destinations:  destinations,
+		Time:          txResponse.Result.Date,
+	}
+
+	return txInfo, nil
+}
+
+// Handle direct transactions (Payment)
+func addSourceDestinationForPayment(txResponse TransactionResponse, sources, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	*sources = append(*sources, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.Amount),
+	})
+
+	*destinations = append(*destinations, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Destination),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.Amount),
+	})
+}
+
+// Handle TakerGets logic ( what the account creating the offer wants to receive )
+func handleTakerGets(txResponse TransactionResponse, sources, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	if txResponse.Result.TakerGets.XRPAmount != "" {
+		appendSourceForTakerGetsXRP(txResponse, sources)
+		appendDestinationsForAffectedNodes(txResponse, destinations, "AccountRoot", 6)
+	} else {
+		amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerGets.TokenAmount.Value)
+		if err == nil {
+			appendSourceForTakerGetsToken(txResponse, sources, amount)
+			appendDestinationsForAffectedNodes(txResponse, destinations, "RippleState", 15)
+		} else {
+			return
+		}
+	}
+}
+
+// Handle TakerPays logic
+func handleTakerPays(txResponse TransactionResponse, sources, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	if txResponse.Result.TakerPays.XRPAmount != "" {
+		appendSourceForTakerPaysXRP(txResponse, destinations)
+		appendSourcesForAffectedNodes(txResponse, sources, destinations, "AccountRoot", 6)
+	} else {
+		amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerPays.TokenAmount.Value)
+		if err == nil {
+			appendDestinationForTakerPaysToken(txResponse, destinations, amount)
+			appendSourcesForAffectedNodes(txResponse, sources, destinations, "RippleState", 15)
+		}
+	}
+}
+
+// Append sources for TakerGets native transactions
+func appendSourceForTakerGetsXRP(txResponse TransactionResponse, sources *[]*xc.LegacyTxInfoEndpoint) {
+	*sources = append(*sources, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerGets.XRPAmount),
+	})
+}
+
+// Append destinations for TakerPays native transactions
+func appendSourceForTakerPaysXRP(txResponse TransactionResponse, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	*destinations = append(*destinations, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerPays.XRPAmount),
+	})
+}
```

Comment:
Again, I fear this is inaccurate?  I thought we parsed the tx metadata directly to get accurate view of transfers.